GoRogue/MapGeneration/Region.cs

```diff
@@ -12,139 +14,91 @@ namespace GoRogue.MapGeneration
-        /// The South-West corner of the region
+        /// Creates a new Polygon, with corners at the provided points
         /// </summary>
-        public Point SouthWestCorner { get; private set; }
+        /// <param name="corners">Each corner of the polygon, which is copied into a new list</param>
+        /// <param name="algorithm">Which Line Algorithm to use</param>
+        /// <param name="components"></param>
+        public Region(IEnumerable<Point> corners, Lines.Algorithm algorithm = Lines.Algorithm.DDA, IComponentCollection? components = null)
+            : this(corners.ToList(), algorithm) { }
 
         /// <summary>
-        /// The North-West corner of the region
+        /// Creates a new Polygon, with corners at the provided points
         /// </summary>
-        public Point NorthWestCorner { get; private set; }
+        /// <param name="corners">The corners of this polygon</param>
+        /// <param name="algorithm">Which Line Algorithm to use</param>
+        /// <param name="components">A collection of components to add to this region</param>
+        public Region(ref List<Point> corners, Lines.Algorithm algorithm = Lines.Algorithm.DDA, IComponentCollection? components = null)
+            : this(corners, algorithm, components) { }
 
         /// <summary>
-        /// the North-East corner of the region
+        /// Returns a new PolygonArea with corners at the provided points.
         /// </summary>
-        public Point NorthEastCorner { get; private set; }
+        /// <param name="algorithm">Which Line-drawing algorithm to use</param>
+        /// <param name="corners">The points which are corners for this polygon</param>
+        public Region(Lines.Algorithm algorithm, params Point[] corners)
+            : this(corners, algorithm, null) { }
 
         /// <summary>
-        /// All points within the region
+        /// Returns a new PolygonArea with corners at the provided points.
         /// </summary>
-        private MultiArea _points;
+        /// <param name="components">A component collection to use for this region</param>
+        /// <param name="corners">The points which are corners for this polygon</param>
+        public Region(IComponentCollection? components = null, params Point[] corners)
+            : this(corners, Lines.Algorithm.DDA, components) { }
 
         /// <summary>
-        /// All of the boundary points of this region.
+        /// Returns a new PolygonArea with corners at the provided points.
         /// </summary>
-        public IReadOnlyArea OuterPoints => _outerPoints;
-        private MultiArea _outerPoints;
+        /// <param name="algorithm">Which Line-drawing algorithm to use</param>
+        /// <param name="components">A component collection for this region</param>
+        /// <param name="corners">The points which are corners for this polygon</param>
+        public Region(Lines.Algorithm algorithm, IComponentCollection? components, params Point[] corners)
```

Comment:
There's a few inconsistencies in how these constructors take parameters; the one above this one takes an _optional_ `components` parameter then `params`; this on takes a non-optional `components` parameter, adds a non-optional `algorithm`.  This is particularly odd to work with since it doesn't quite mimic the arrangement of the `PolygonArea` constructors.

I know that particularly when `params` are involved, there are some limitations to optional parameters and their usefulness; but ideally, we could mimic the `PolygonArea` constructors as closely as possible (obviously adding the component parts).  Thoughts?